### ISSUE TITLE
torch.nn.functional.log_softmax  parameter '_stacklevel' undocumented

### ISSUE THREAD

**Issue by cheyennee** [pytorch/pytorch#83163]
### 📚 The doc issue

The documentation of the very popular torch.nn.functional.log_softmax doesn't explain what _stacklevel is, why it's set to 3 and what to do with it.

**Comment by vadimkantorov** [comment]
Related: https://github.com/pytorch/pytorch/issues/36524

**Comment by cheyennee** [comment]
> Related: #36524. As discussed there, this `_stacklevel` is a remnant of migrating to explicit-only `dim` and to control warnings about not setting `dim` and the default value used. I think since a long time ago, this `_stacklevel` argument should be deleted.

I want to know whether the value of '_stacklevel' can set to string❓ Even value in documentation is set to 3, I found that when it's set to string, such as "max", these two API works normally.

```
import torch
results={}
arg_1 = torch.rand([4, 12, 141, 141], dtype=torch.float32)
arg_2 = -1
arg_3 = "max"
results['res'] = torch.nn.functional.softmax(arg_1,arg_2,_stacklevel=arg_3,)
```

**Comment by cheyennee** [comment]
> I think the essence of the story is that _stacklevel is passed to some `warnings.warn(...)` function or even is unused now / cast implicitly. So it's not very interesting to inquire about its possible values. If I'm correct, this argument should be dropped entirely from arguments list, and this is the right course of action about _stacklevel
> 
> But hopefully we can hear from devs @albanD @ezyang

Get it. I think you're right, because just now I found that bool / list or other value could be assigned to _stacklevel. Thank u🥰

**Comment by ezyang** [comment]
Let's drop it

**Comment by albanD** [comment]
This argument is only used for the depraction warning raised when `dim=None`. Otherwise it is not used and its type is not checked indeed.

Since `dim=None` has been deprecated for 5+ years now (https://github.com/pytorch/pytorch/commit/98e67448fa78bd1bc6f05920ad03efceecc10066), I think it is ok to remove it (and remove the `_stacklevel` arg)

**Comment by vadimkantorov** [comment]
After dropping, probably even functional wrapper can be dropped or auto-generated instead?
